api/subjects/serializers.py

```diff
@@ -0,0 +1,117 @@
+from rest_framework import serializers as ser
+from rest_framework import exceptions
+
+from framework.auth.core import Auth
+from framework.exceptions import PermissionsError
+
+from api.base.serializers import (
+    JSONAPISerializer,
+    JSONAPIRelationshipSerializer,
+    LinksField,
+    RelationshipField,
+    BaseAPISerializer,
+)
+from osf.exceptions import NodeStateError, ValidationValueError
+
+
+class UpdateSubjectsMixin(object):
+    def update_subjects_method(self, resource, subjects, auth):
+        # Method to update subjects on resource
+        raise NotImplementedError()
+
+    def update_subjects(self, resource, subjects, auth):
+        """Updates subjects on resource and handles errors.
+
+        :param object resource: Object for which you want to update subjects
+        :param list subjects: Subjects array (or array of arrays)
+        :param object Auth object
+        """
+        try:
+            self.update_subjects_method(resource, subjects, auth)
+        except PermissionsError as e:
+            raise exceptions.PermissionDenied(detail=str(e))
+        except ValueError as e:
+            raise exceptions.ValidationError(detail=str(e))
+        except ValidationValueError as e:
+            raise exceptions.ValidationError(detail=e[0])
+        except NodeStateError as e:
+            raise exceptions.ValidationError(detail=str(e))
+
+
+class SubjectSerializer(JSONAPISerializer):
+    filterable_fields = frozenset([
+        'text',
+        'parent',
+        'id',
+    ])
+    id = ser.CharField(source='_id', required=True)
+    text = ser.CharField(max_length=200)
+    share_title = ser.CharField(source='provider.share_title', read_only=True)
+
+    parent = RelationshipField(
```

Comment:
Would it be reasonable to put `always_embed=True` on `parent`? The new subject picker is supposed to allow searching for subjects (we'll use `?filter[text]=foo`), and we typically want to display them with their full parent lineage. It would be nice to avoid a second request to get that lineage, and `embed=parent` only goes one level deep.

And come to think of it, for that search we'll need to filter on `share_title`... could that be added to `filterable_fields`?

Reply:
See https://github.com/CenterForOpenScience/osf.io/pull/8948#discussion_r275141028 - change required in order to add always_embed=True for parents